### PR TITLE
Tap if necessary during unlink_conflicts

### PR DESCRIPTION
**Pull request by scpeters**
Related to #86, rescue from TapFormulaUnavailableError
during call to formula.recursive_dependencies
in unlink_conflicts (called by install_bottled_dependent)
and add the missing tap.

I found #86 when trying to `test-bot` a formula (ie. `osrf/simulation/gazebo7`) that depends on an untapped formula (ie. `cartr/qt4/qt@4`). For this PR, I found a problem when rebuilding a bottle for `ignition-transport`, which is a dependency of `gazebo7`. When trying to `install_bottled_dependent`, there was a call to `formula.recursive_dependencies` in `unlink_conflics` ([stacktrace here](https://gist.github.com/scpeters/7b3f46e672cd126507e06aab19e99030)).

### CODE REVIEW

**Comment by scpeters** [comment]
Travis is unhappy with the `brew tests` invocations in the OS X build, but I think that is unrelated to this PR?

File: cmd/brew-test-bot.rb [hunk 1]
```diff
@@ -532,7 +532,15 @@ def unlink_conflicts(formula)
       return if formula.keg_only?
       return if formula.linked_keg.exist?
       conflicts = formula.conflicts.map { |c| Formulary.factory(c.name) }.select(&:installed?)
-      formula.recursive_dependencies.each do |dependency|
+      begin
+        formula_recursive_dependencies = formula.recursive_dependencies
```
Comment (MikeMcQuaid) [inline cmd/brew-test-bot.rb]: I'd suggest
```ruby
formula_recursive_dependencies = begin
  formula.recursive_dependencies
```

**Comment by MikeMcQuaid** [comment]
> Travis is unhappy with the brew tests invocations in the OS X build, but I think that is unrelated to this PR?

I'm rerunning both this build and the `master` build.

**Comment by scpeters** [comment]
Good idea. I'm offline for the weekend but will make the change when I get
back

On May 26, 2017 7:02 AM, "Mike McQuaid" <notifications@github.com> wrote:

*@MikeMcQuaid* requested changes on this pull request.
------------------------------

In cmd/brew-test-bot.rb
<https://github.com/Homebrew/homebrew-test-bot/pull/91#discussion_r118708561>
:

> @@ -532,7 +532,15 @@ def unlink_conflicts(formula)
       return if formula.keg_only?
       return if formula.linked_keg.exist?
       conflicts = formula.conflicts.map { |c|
Formulary.factory(c.name) }.select(&:installed?)
-      formula.recursive_dependencies.each do |dependency|
+      begin
+        formula_recursive_dependencies = formula.recursive_dependencies

I'd suggest

formula_recursive_dependencies = begin
  formula.recursive_dependencies

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/Homebrew/homebrew-test-bot/pull/91#pullrequestreview-40528483>,
or mute the thread

**Comment by MikeMcQuaid** [comment]
@scpeters I think either this PR broke something or it needs rebased on `master`.

**Comment by scpeters** [comment]
I rebased on master and squashed in your recommended syntax change

**Comment by scpeters** [comment]
travis is happy now

**Comment by MikeMcQuaid** [comment]
Thanks again @scpeters!

**Pull request merged**